recipe/patches/0001-Don-t-build-test-protos.patch

```diff
@@ -0,0 +1,425 @@
+From cbf274f8cf9693eced4d177503062c457df107d8 Mon Sep 17 00:00:00 2001
+From: "Uwe L. Korn" <uwe.korn@quantco.com>
+Date: Wed, 27 Nov 2024 19:12:48 +0100
+Subject: [PATCH] Don't build test protos
+
+---
+ CMakeLists.txt | 405 -------------------------------------------------
+ 1 file changed, 405 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index bbd536c..8aa66f7 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -890,411 +890,6 @@ protobuf_generate_grpc_cpp_with_import_path_correction(
+ protobuf_generate_grpc_cpp_with_import_path_correction(
+   test/core/tsi/alts/fake_handshaker/transport_security_common.proto test/core/tsi/alts/fake_handshaker/transport_security_common.proto
+ )
+-protobuf_generate_grpc_cpp_with_import_path_correction(
+-  third_party/envoy-api/envoy/annotations/deprecation.proto envoy/annotations/deprecation.proto
+-)
```

Comment:
What's the pattern to identify these as "test" protos (in contrast to the one further up under `test/`)? The fact that it's under `third_parthy`?

Can you describe what the problem with them is?

Reply:
They are actually all only used for testing. The ones I have removed here were newly added in the 1.83 release and require the existence of various git submodules. I will try to submit an upstream patch to guard them in `GRPC_BUILD_TESTS`.